spec/services/nla/citations/wikimedia_citation_service_spec.rb

```diff
@@ -121,22 +121,40 @@
     end
   end
 
-  describe "#build_pi" do
-    context "when there is a pi" do
+  describe "#build_persistent_url" do
+    context "when there is an id" do
       let(:document) { SolrDocument.new(marc_ss: book_marc, id: "123", title_tsim: "Title", format: ["Book"]) }
 
-      it "returns the pi" do
-        allow(document).to receive(:pi).and_return(["https://nla.gov.au/nla.obj-234175885/flightdiagram"])
+      it "returns the id" do
+        expect(service.build_persistent_url).to eq(" | url=https://nla.gov.au/nla.cat-vn123\n")
+      end
+    end
+
+    context "when there is no id" do
+      let(:document) { SolrDocument.new(marc_ss: book_marc, title_tsim: "Title", format: ["Book"]) }
 
-        expect(service.build_pi).to eq(" | url=https://nla.gov.au/nla.obj-234175885/flightdiagram\n")
+      it "returns nil" do
+        expect(service.build_persistent_url).to be_nil
       end
     end
+  end
 
-    context "when there is no pi" do
+  describe "#build_access_date" do
+    context "when date is correct" do
```

Comment:
You should use `freeze_time` or `travel_to` to test against a consistent date or time. https://api.rubyonrails.org/v5.2.4/classes/ActiveSupport/Testing/TimeHelpers.html#method-i-travel_to